src/pbt/v2/client/airflow_client/composer_client.py

```diff
@@ -0,0 +1,183 @@
+        response.raise_for_status()
+        response_data = response.json()
+
+        # Parse JSON into Dag class
+        dag = DAG(**response_data)
+        return dag
+
+    def delete_dag(self, dag_id: str) -> str:
+        response = requests.request('DELETE', f"{self.airflow_url}/api/v1/dags/{dag_id}")
+        response.raise_for_status()
+        return response.text
+
+    def _create_key_json_file(self, key_json: str) -> str:
+        temp_file = tempfile.NamedTemporaryFile(prefix="composer", suffix=".json", delete=False)
+
+        # Write content to the temporary file
+        Path(temp_file.name).write_text(key_json)
+
+        return temp_file.name
+
+    def _get_dag_location(self, gcp_path_info: GCSPathInfo, dag_id: str) -> str:
+        if gcp_path_info is not None and gcp_path_info.path is not None:
+            return f"{gcp_path_info.path}/{dag_id}.zip"
+        else:
+            print(f"Shouldn't be here: {gcp_path_info}")
+            return f"{dag_id}.zip"
+
+    def _get_secrets_manager(self):
+        # fill up how to get the credentials
+        credentials = ServiceAccountCredentials.from_service_account_file(self.key_json_file)
+        client = secretmanager.SecretManagerServiceClient(credentials=credentials)
+        return client
+
+    def _get_gcs_path_info(self) -> GCSPathInfo:
+        pattern = re.compile(r"^g[c]?s://([a-z0-9][-a-z0-9.]*[a-z0-9]):?/(.*)?$")
+        match = pattern.match(self.dag_location)
+
+        if match:
+            bucket, path = match.groups()
+            gcs_path_info = GCSPathInfo(bucket, "/".join(path.split("/")))
+        else:
+            gcs_path_info = None
+
+        return gcs_path_info
+
+    def _get_authenticated_session(self):
+
+        if self.client_id is not None and len(self.client_id) > 0:
+            print(f'Client Id {self.client_id}')
+            credentials = service_account.IDTokenCredentials \
+                .from_service_account_info(json.loads(self.key_json_file)) \
+                .with_target_audience(self.client_id) \
+                .with_scopes(self._SCOPES)
+
+
+        else:
+            credentials = service_account.Credentials.from_service_account_info(json.loads(self.key_json_file)) \
+                .with_scopes(self._SCOPES)
+        return AuthorizedSession(credentials)
+
+    @retry((HTTPError,), total_tries=5, delay_in_seconds=10, backoff=0)
+    def _set_pause_state(self, dag_id: str, is_paused: bool) -> DAG:
+        session = self._get_authenticated_session()
+        response = session.request(method='PATCH', url=f"{self.airflow_url}/api/v1/dags/{dag_id}",
+                                   data=json.dumps({"is_paused": is_paused}),
+                                   headers={'Content-Type': 'application/json', 'Accept': 'application/json'})
+        response.raise_for_status()
+        response_data = response.json()
+        return DAG.create(response_data)
```

Comment:
are `DAG.create(x)` and `DAG(**x)` same?

Reply:
this structure is nested thereby flat mapping didn't work.